client/post-editor/editor-deprecation-dialog/index.jsx

```diff
@@ -0,0 +1,134 @@
+/**
+ * External dependencies
+ */
+
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+/**
+ * WordPress dependencies
+ */
+import { Button, Modal } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import isEditorDeprecationDialogShowing from 'state/selectors/is-editor-deprecation-dialog-showing';
+import { hideEditorDeprecationDialog } from 'state/ui/editor-deprecation-dialog/actions';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { setSelectedEditor } from 'state/selected-editor/actions';
+import { localize } from 'i18n-calypso';
+import { composeAnalytics, recordTracksEvent, withAnalytics } from 'state/analytics/actions';
+import { getEditorPostId } from 'state/ui/editor/selectors';
+import { getEditedPostValue } from 'state/posts/selectors';
+import getGutenbergEditorUrl from 'state/selectors/get-gutenberg-editor-url';
+
+/**
+ * Style dependencies
+ */
+import '@wordpress/components/build-style/style.css';
+import './style.scss';
+
+class EditorDeprecationDialog extends Component {
+	static propTypes = {
+		// connected properties
+		translate: PropTypes.func,
+		gutenbergUrl: PropTypes.string,
+		hideDialog: PropTypes.func,
+		optIn: PropTypes.func,
+		logNotNow: PropTypes.func,
+		siteId: PropTypes.number,
+		isDialogShowing: PropTypes.bool,
+	};
+
+	notNow = () => {
+		const { logNotNow, hideDialog } = this.props;
+		logNotNow();
+		hideDialog();
+	};
+
+	optInToGutenberg = () => {
+		const { gutenbergUrl, optIn, siteId, hideDialog } = this.props;
+		hideDialog();
+		optIn( siteId, gutenbergUrl );
+	};
+
+	render() {
+		const { isDialogShowing, translate } = this.props;
+
+		if ( ! isDialogShowing ) {
+			return null;
+		}
+
+		return (
+			<Modal
```

Comment:
We also have a <Dialog> component. I'm not sure why we have two, or which one to use...

Reply:
Oh I see `Modal` comes from core, so that's probably the better choice :)